pkg/controller/predicates/predicate.go

```diff
@@ -0,0 +1,127 @@
+package predicates
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/nukleros/operator-builder-tools/pkg/controller/workload"
+	"github.com/nukleros/operator-builder-tools/pkg/resources"
+)
+
+// ResourcePredicates returns the filters which are used to filter out the common reconcile events
+// prior to reconciling the child resource of a component.
+func ResourcePredicates(r workload.Reconciler, req *workload.Request) predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return needsReconciliation(
+				r,
+				req,
+				e.ObjectOld,
+				e.ObjectNew,
+			)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			// do not run reconciliation on unknown events
+			return false
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			// do not run reconciliation again when we just created the child resource
+			return false
+		},
+	}
+}
+
+// WorkloadPredicates returns the filters which are used to filter out the common reconcile events
+// prior to reconciling an object for a component.
+func WorkloadPredicates() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return !e.DeleteStateUnknown
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+}
+
+// needsReconciliation performs some simple checks and returns whether or not a
+// resource needs to be updated.
+func needsReconciliation(r workload.Reconciler, req *workload.Request, existing, requested client.Object) bool {
+	// skip if the resources versions are the same
+	if existing.GetResourceVersion() == requested.GetResourceVersion() {
+		return false
+	}
+
+	// skip if the objects support observed generation and they are equal
```

Comment:
I believe this will reintroduce a bug that we fixed here:

https://github.com/vmware-tanzu-labs/operator-builder/commit/bd14abb50a1e46de0fdf016137e1c6af1f70286e#diff-0f0bc1ea96d2116bd8394fd9b1110f4400a8e89ce1127db3aefcb25f2aba039eL112